### PR TITLE
Ubuntu/noble: fail fast on network-config

### PR DESCRIPTION
**Pull request by TheRealFalcon**
See https://github.com/canonical/cloud-init/issues/5753

If approved, will also apply to jammy and focal

### CODE REVIEW

**Comment by a-dubs** [comment]
How come the packaging CI action failed...? :thinking:

**Comment by TheRealFalcon** [comment]
The branch needs an upstream snapshot (rebase essentially), but I don't want to do that here.

File: debian/patches/fast-nocloud-network.patch [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+-        if network_resp.ok():
+-            network = load_yaml(network_resp.contents)
+     md_resp = url_helper.read_file_or_url(
+         md_url, timeout=timeout, retries=retries
+     )
+@@ -1058,6 +1048,18 @@ def read_seeded(base="", ext="", timeout
+         else:
+             LOG.debug("Error in vendor-data response")
+
++    network = None
```
Comment (holmanb) [inline debian/patches/fast-nocloud-network.patch:32]: Was there a reason to reorder the network call later?

File: debian/patches/fast-nocloud-network.patch [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
++    if md or ud or vd:
++        try:
++            network_resp = url_helper.read_file_or_url(
++                network_url, timeout=timeout, retries=0
```
Comment (holmanb) [inline debian/patches/fast-nocloud-network.patch:36]: This is going to add a 5s timeout to every NoCloud server which doesn't provide a network-config. That sounds like change in a change in behavior that would negatively affect some users.

File: debian/patches/fast-nocloud-network.patch (hunk 1 above)
Comment (TheRealFalcon) [reply debian/patches/fast-nocloud-network.patch:32]: To only make the network call if any of the other calls succeeded.

File: debian/patches/fast-nocloud-network.patch (hunk 2 above)
Comment (TheRealFalcon) [reply debian/patches/fast-nocloud-network.patch:36]: Possible, but I don't think that will be true in most cases. Any reasonable server will return a 404 immediately, and with retries=0, that should finish fairly quick. In the case of a firewall swallowing packets, the previous calls should fail first, preventing this one from running.

Comment (holmanb) [reply debian/patches/fast-nocloud-network.patch:36]: > Possible, but I don't think that will be true in most cases.

Sure, agreed.

> Any reasonable server will return a 404 immediately, and with retries=0, that should finish fairly quick.

I don't disagree - for a reasonable server this should probably be just fine. I'm just concerned about user infrastructure that doesn't behave reasonably. Is "anything broken by this change is unreasonable" reasonable justification for shipping a change in behavior?

**Comment by TheRealFalcon** [comment]
After more discussion, this represents too much risk. Closing

**Pull request closed without merging**